app/components/OnboardingComponent.tsx

```diff
@@ -0,0 +1,143 @@
+'use client';
+
+import WelcomeScreen from '@/app/components/WelcomeScreen';
+import { useState } from 'react';
+import { Box } from '@mui/material';
+import OnboardingSignup from '@/app/components/OnboardingSignup';
+import OnboardingNotification from '@/app/components/OnboardingNotification';
+import OnboardingSites from '@/app/components/OnboardingSites';
+import OnboardingAddNewSite from '@/app/components/OnboardingAddNewSite';
+import { useRouter } from 'next/navigation';
+
+interface OnboardingProps {
+  dict: Dict;
+}
+
+type UserData = {
+  name: string;
+  phoneNumber: string;
+  password: string;
+};
+
+export interface UserFormErrorData extends UserData {
+  name: string;
+  phoneNumber: string;
+  password: string;
+  confirmPassword: string;
+}
+
+export interface UserFormData extends UserData {
+  confirmPassword: string;
+  allowPushNotifications: boolean;
+  sites: SiteData[];
+}
+
+export interface SiteData {
+  name: string;
+  type: string;
+  location: string; //Don't know exactly what type this needs to be. Might need to be both lan and lat
+}
+
+const initialValues: UserFormData = {
```

Comment:
Could all of these initial*-objects ble defined inline instead? Or are they used more than one place?